### PR TITLE
CANPacker: refactor to avoid undefined signals and set default value for  the unspecified signals.

### PR DESCRIPTION
**Pull request by deanlee**
@sshane any thoughts on this implementation?

resolve https://github.com/commaai/opendbc/issues/500 https://github.com/commaai/opendbc/issues/882

### CODE REVIEW

File: can/packer.cc [hunk 1]
```diff
@@ -45,24 +45,23 @@ std::vector<uint8_t> CANPacker::pack(uint32_t address, const std::vector<SignalP
 
   // set all values for all given signal/value pairs
   bool counter_set = false;
-  for (const auto& sigval : signals) {
-    auto sig_it = signal_lookup.find(std::make_pair(address, sigval.name));
-    if (sig_it == signal_lookup.end()) {
-      // TODO: do something more here. invalid flag like CANParser?
-      WARN("undefined signal %s - %d\n", sigval.name.c_str(), address);
-      continue;
```
Comment (sshane) [inline can/packer.cc:53]: we should make sure to preserve the warning for specifying non-existent signals, or better yet, make it apparent you're doing something weird: https://github.com/commaai/opendbc/issues/500

**Comment by sshane** [comment]
anything you can think of to split into separate changes/PRs?

**Comment by sshane** [comment]
We can't merge this just yet, as is, since certain places in openpilot rely on this behavior. For example, openpilot zeros the bytes for the gas interceptor command, which when you set the values to 0, you get non-zero bytes

image

Previous parsed values (supposed to be 0):

{'COUNTER_PEDAL': 9.0,
 'CHECKSUM_PEDAL': 132.0,
 'GAS_COMMAND': -75.555,
 'GAS_COMMAND2': -151.111}

And now they're

{'COUNTER_PEDAL': 9.0,
 'CHECKSUM_PEDAL': 86.0,
 'GAS_COMMAND': -0.01125000000001819,
 'GAS_COMMAND2': -0.023500000000012733}

**Comment by deanlee** [comment]
I see. closed it for now

**Pull request closed without merging**